packages/commonwealth/client/scripts/views/layout.tsx

```diff
@@ -48,125 +64,169 @@ const shouldDeferChain = ({ deferChain }: ShouldDeferChainAttrs) => {
+    selectedScope,
+    scopeToLoad,
+    scopeIsEthereumAddress,
+    scopeMatchesChain,
+    shouldDeferChain,
+  ]);
+
+  useNecessaryEffect(() => {
+    // IFB 6: If deferChain is false on the page we’re routing to, but we
+    // have loaded with isChainDeferred=true (previously from step 5),
+    // then call initChain and render a LoadingLayout immediately.
+    if (selectedScope && isChainDeferred && !shouldDeferChain) {
+      setIsLoading(true);
+      setIsChainDeferred(false);
+      initChain().finally(() => {
+        setIsLoading(false);
       });
-      return <LoadingLayout />;
     }
-
-    if (!scope && app.chain && app.chain.network) {
-      // Handle the case where we unload the network or community, if we're
-      // going to a page that doesn't have one
-      // Include this in if for isCustomDomain, scope gets unset on redirect
-      // We don't need this to happen
-      if (!app.isCustomDomain()) {
-        deinitChainOrCommunity().then(() => {
-          this.loadingScope = null;
-          redraw();
-        });
-      }
-      return <LoadingLayout />;
+  }, [selectedScope, isChainDeferred, shouldDeferChain]);
+
+  useNecessaryEffect(() => {
+    // IFB 7: If scope is not defined (and we are not on a custom domain),
+    // deinitialize whatever chain is loaded by calling deinitChainOrCommunity,
+    // then set loadingScope to null. Render a LoadingLayout immediately.
+    if (
+      !selectedScope &&
+      !app.isCustomDomain() &&
+      app.chain &&
+      app.chain.network
+    ) {
+      setIsLoading(true);
+      deinitChainOrCommunity().finally(() => {
+        setScopeToLoad(null);
+        setIsLoading(false);
+      });
     }
+  }, [selectedScope]);
 
+  // IFB 1: If initApp() threw an error, show application error.
+  if (app.loadingError) {
+    return <ApplicationError />;
+  }
+
+  // Show loading state for these cases
+  // -
+  // IFB 2: If initApp() hasn’t finished loading yet
+  // -
+  // IFB 3: If the user has navigated to an ethereum address directly,
+  // init a new token chain immediately and show loading state
+  // -
+  // IFB 5: If scope is different from app.activeChainId() at render
+  // time (and we are not loading another community at the same time,
+  // via this.loadingScope), set this.loadingScope to the provided scope,
+  // and then call selectChain, passing deferChain through. If deferChain
+  // is false once selectChain returns, call initChain. Render a LoadingLayout
+  // immediately (before selectChain resolves).
+  // -
+  // IFB 6: If deferChain is false on the page we’re routing to, but we
+  // have loaded with isChainDeferred=true (previously from step 5),
+  // then call initChain and render a LoadingLayout immediately.
+  // -
+  // IFB 7: If scope is not defined (and we are not on a custom domain),
+  // deinitialize whatever chain is loaded by calling deinitChainOrCommunity,
+  // then set loadingScope to null. Render a LoadingLayout immediately.
+  if (
```

Comment:
Will this case render properly if e.g. isLoading changes?

Reply:
Yes, it has less to do with isLoading and more with `scope` related stuff

Reply:
I verified by first going to `/dydx` directly and then using click to go back to `/for-you` page and the opposite